packages/sdk/src/core/teamsfx.browser.ts

```diff
@@ -0,0 +1,117 @@
+import { UserInfo } from "../models/userinfo";
+import { formatString } from "../util/utils";
+import { ErrorWithCode, ErrorCode, ErrorMessage } from "../core/errors";
+import { internalLogger } from "../util/logger";
+import { TeamsFxConfiguration } from "../models/teamsfxConfiguration";
+
+/**
+ * A class providing credential and configuration.
+ * @beta
+ */
+export class TeamsFx implements TeamsFxConfiguration {
+  private configuration: Map<string, string | undefined>;
+  private teamsUserCredential?: TeamsUserCredential;
+  public identityType: IdentityType;
+
+  constructor(identityType?: IdentityType) {
+    this.identityType = identityType ?? IdentityType.User;
+    if (this.identityType !== IdentityType.User) {
+      const errorMsg = formatString(
+        ErrorMessage.IdentityTypeNotSupported,
+        this.identityType.toString(),
+        "TeamsFx"
+      );
+      internalLogger.error(errorMsg);
+      throw new ErrorWithCode(errorMsg, ErrorCode.IdentityTypeNotSupported);
+    }
+    this.configuration = new Map<string, string>();
+    this.loadFromEnv();
+  }
+
+  private loadFromEnv(): void {
+    let env: any;
+    if (window && (window as any).__env__) {
+      // testing purpose
+      env = (window as any).__env__;
+    } else {
+      env = process.env;
+    }
+    if (!env) {
+      return;
+    }
+    this.configuration.set("authorityHost", env.REACT_APP_AUTHORITY_HOST);
+    this.configuration.set("tenantId", env.REACT_APP_TENANT_ID);
+    this.configuration.set("clientId", env.REACT_APP_CLIENT_ID);
+    this.configuration.set("initiateLoginEndpoint", env.REACT_APP_START_LOGIN_PAGE_URL);
+    this.configuration.set("applicationIdUri", env.M365_APPLICATION_ID_URI);
+    this.configuration.set("apiEndpoint", env.REACT_APP_FUNC_ENDPOINT);
+    this.configuration.set("apiName", env.REACT_APP_FUNC_NAME);
+
+    Object.keys(env).forEach((key: string) => {
+      const value = env[key];
+      if (key.startsWith("TEAMSFX_") && value) {
+        this.configuration.set(key.substring(8), value);
+      }
+    });
+  }
+
+  public getCredential(): TokenCredential {
+    if (!this.teamsUserCredential) {
+      this.teamsUserCredential = new TeamsUserCredential(Object.fromEntries(this.configuration));
+    }
+    return this.teamsUserCredential;
+  }
+
+  public async getUserInfo(): Promise<UserInfo> {
+    return await (this.getCredential() as TeamsUserCredential).getUserInfo();
+  }
+
+  public async login(scopes: string | string[]): Promise<void> {
+    await (this.getCredential() as TeamsUserCredential).login(scopes);
+  }
+
+  public setSsoToken(ssoToken: string): TeamsFx {
+    return this;
+  }
+
+  public setCustomConfig(customConfig: Record<string, string>): TeamsFx {
+    for (const key of Object.keys(customConfig)) {
+      const value = customConfig[key];
+      if (value) {
+        this.configuration.set(key, value);
+      }
+    }
+    this.teamsUserCredential = undefined;
```

Comment:
Yes this could be a problem and what I think is to hide the direct usage of credential instance. Since `TeamsFx:getCredential()` returns `TokenCredential` which doesn't have `login()` function.
That's the reason to expose `TeamsFx:login()` to do user consent process. One possible fix is to keep the internal credential instance and update config of credential by adding a new interface like `TeamsUserCredential:updateConfig()`.

Reply:
Fixed using second option.